apis/v1beta1/types.go

```diff
@@ -370,6 +370,10 @@ type NetworkDeviceSpec struct {
 	// For more information see the netplan reference (https://netplan.io/reference#dhcp-overrides)
 	// +optional
 	DHCP6Overrides *DHCPOverrides `json:"dhcp6Overrides,omitempty"`
+
+	// IPConfigurationNotMandatory allows for the device to not have IP address or DHCP configured.
+	// +optional
+	IPConfigurationNotMandatory bool `json:"ipConfigurationNotMandatory,omitempty"`
```

Comment:
maybe we could use something a little more explicit here, e.g. `skipWaitForAddress`?

Reply:
I think we should go into more detail at the godoc.

If this new variable gets set to true, this device gets ignored when checking for an existing ip address. And propably that at least one device should get an IP address (dunno if this is true but I can think of provisioning to fail at different places if there is no IP here)?

`SkipWaitForAddress` sounds better to me but not yet perfect. No strong opinion on the name or a better recommendation from my side though...

What about: `SkipIPAllocation` or `IgnoreIPAllocation`

Reply:
`SkipIPAllocation` or `IgnoreIPAllocation` sounds more straightforward to me. And I'm thinking perhaps adding `static` is more precise, like `SkipStaticIPAllocation` or `IgnoreStaticIPAllocation` as we're only skipping the static IP allocation but not the DHCP allocation.